File: algorithm/javascript/algorithm/two_pointer/check_for_target.js
const check_for_target = (nums, target) => {
    let left = 0;
    let right = nums.length - 1;

    while(left < right) { 
      let curr = nums[left] + nums[right];
      if (curr == target){
        return true;
      }

      if (curr > target){
        left++;
      }

      if (curr < target){
        right--;
      }
    }

    return false;

  };
  
  console.log(check_for_target([1,2,3,4,5], 6)); // returns [3, 4, 7, 10, 12]
  console.log(check_for_target([1,2,3,4,5], 10)); // returns [3, 4, 7, 10, 12]

File: algorithm/python/algorithm/two_pointer/check_for_target.py
def check_for_target(numbers, target):
    left =0
    right = len(numbers)-1
    while left < right:
        curr = numbers[left] + numbers[right]
        if curr == target:
            return True
        if curr > target:
            left += 1
        if curr < target:
            right -= 1
    return False

print(check_for_target([1,2,3,4,5,6,7,8,9], 10))
print(check_for_target([1,2,3,4,5,6,7,8,9], 20))